File: SMILE/tutorials/python/hello.py
import pysmile

# pysmile_license is your license key
import pysmile_license

def hello_smile():
    net = pysmile.Network()
    net.read_file("VentureBN.xdsl");
        
    net.set_evidence("Forecast", "Moderate")
    net.update_beliefs()

    beliefs = net.get_node_value("Success")
    for i in range(0, len(beliefs)):
        print(net.get_outcome_id("Success", i) + "=" + str(beliefs[i]))

hello_smile()

File: SMILE/tutorials/cpp/hello.cpp
// hello.cpp

#include <cstdio>
#include <smile.h>
#include "smile_license.h" // your licensing key

int main()
{
    DSL_errorH().RedirectToFile(stdout);
    DSL_network net;
    int res = net.ReadFile("VentureBN.xdsl");
    if (DSL_OKAY != res)
    {
        return res;
    }

    int handle = net.FindNode("Forecast");
    if (handle < 0)
    {
        return handle;
    }

    DSL_node *f = net.GetNode(handle);
    int idx =
        f->Definition()->GetOutcomesNames()->FindPosition("Moderate");
    if (idx < 0)
    {
        return idx;
    }
    f->Value()->SetEvidence(idx);

    net.UpdateBeliefs();

    handle = net.FindNode("Success");
    if (handle < 0)
    {
        return handle;
    }
    DSL_node *s = net.GetNode(handle);
    const DSL_Dmatrix &beliefs = *s->Value()->GetMatrix();
    const DSL_idArray &outcomes =
        *s->Definition()->GetOutcomesNames();
    for (int i = 0; i < outcomes.NumItems(); i++)
    {
        printf("%s=%g\n", outcomes[i], beliefs[i]);
    }

    return DSL_OKAY;
}
